airbyte-workers/src/main/java/io/airbyte/workers/process/KubeProcessBuilderFactory.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.workers.process;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.workers.WorkerException;
+import java.io.FileWriter;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KubeProcessBuilderFactory implements ProcessBuilderFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KubeProcessBuilderFactory.class);
+
+  private static final Path DATA_MOUNT_DESTINATION = Path.of("/data");
+  private static final String IMAGE_EXISTS_SCRIPT = "image_exists.sh";
+
+  private final Path workspaceRoot;
+
+  public KubeProcessBuilderFactory(Path workspaceRoot) {
+    this.workspaceRoot = workspaceRoot;
+  }
+
+  @Override
+  public ProcessBuilder create(final Path jobRoot, final String imageName, final String... args) throws WorkerException {
```

Comment:
in a better world, should we be passing job and attempt id in here? i appreciate that it is a hassle given the current setup and will explode the scope of this PR so i'm okay with it not happening right this second. if so then we should at least create the issue to pass this information through, and  we should probably do it soon  this implicit version of finding job id and attempt id is pretty brittle.

Reply:
Yeah this definitely isn't ideal. It also isn't too difficult to add, I'll do that.